File: src/python/Regexp1.py
#
# Regexp1.py
#
# Created by Jukka Aro on 13.3.2012.
#

import sys
import re

def regexp_match(input_file, or_operator, dry_run):
	f = open(input_file, 'r')
	if dry_run:
		for line in f:
			pass
	else:
		if or_operator:
			prog = re.compile(r"\d+|\w+\d+\.")
		else:
			prog = re.compile(r"\d+")
		cnt = 0
		for line in f:
			if(prog.search(line) != None):
				cnt += 1
	f.close()


# program entry:

if len(sys.argv) < 2 or len(sys.argv) > 4:
	print "Illegal number of arguments"
	sys.exit()

or_operator = False
dry_run = False

if len(sys.argv) == 3:
	if sys.argv[2] == '--with-or-operator':
		or_operator = True
	elif sys.argv[2] == '--dry-run':
		dry_run = True
	else:
		print "Illegal argument: " + sys.argv[2]
		sys.exit()
if len(sys.argv) == 4:
	if sys.argv[2] == '--with-or-operator' and sys.argv[3] == '--dry-run':
		or_operator = True
		dry_run = True
	else:
		print "Illegal arguments: " + sys.argv[2] + " " + sys.argv[3]
		sys.exit()
	
regexp_match(sys.argv[1], or_operator, dry_run)

File: src/perl/Regexp1.pl
#
# Regexp1.pl
#
# Created by Jukka Aro on 14.3.2012.
#

sub regexp_match{
	open FILE, $_[0] or die $!;
	if($_[2] == 0){
		if($_[1] == 1){
			$regexp = qr/\d+|\w+\d+\./;
		}
		else{
			$regexp = qr/\d+/;
		}
		$count = 0;
		while(<FILE>){
			if(/$regexp/){
				$count++;
			}
		}
	}
	else{
		while(<FILE>){}
	}
	close(FILE);
}

#program entry:

if($#ARGV > 2){
	die "Illegal number of arguments";
}

$or = 0;
$dry_run = 0;

if($#ARGV == 1){
	if($ARGV[1] eq "--with-or-operator"){
		$or = 1;
	}
	elsif($ARGV[1] eq "--dry-run"){
		$dry_run = 1;
	}
	else{
		die "Illegal argument: ".$ARGV[1];
	}
}
if($#ARGV == 2){
	if($ARGV[1] eq "--with-or-operator" && $ARGV[2] eq "--dry-run"){
		$or = 1;
		$dry_run = 1;
	}
	else{
		die "Illegal arguments: ".$ARGV[1]." ".$ARGV[2];
	}
}

regexp_match($ARGV[0], $or, $dry_run);
